src/vs/gitpod/browser/gitpodInsightsAppender.ts

```diff
@@ -40,42 +77,51 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
 		}
 	}
 
-	private sendAnalytics(eventName: string, data: any): void {
+	private async sendAnalytics(eventName: string, data: any): Promise<void> {
 		try {
-			const trackMessage = mapTelemetryData('window', eventName, data);
-			if (!trackMessage) {
+			const mappedEvent = await this.mapAnalytics(eventName, data);
+			if (!mappedEvent) {
 				return;
 			}
-			trackMessage.properties = {
-				...trackMessage.properties,
-				...this._baseProperties,
-			};
-			if (this.devMode) {
-				if (this.productService.gitpodPreview?.log?.analytics) {
-					console.log('Gitpod Analytics: ', JSON.stringify(trackMessage, undefined, 2));
-				}
-			} else {
-				// TODO(ak) get rid of it
-				// it is bad usage of window.postMessage
-				// we should use Segment directly here and publish to production/staging untrusted
-				// use server api to resolve a user
-				window.postMessage({ type: 'vscode_telemetry', event: trackMessage.event, properties: trackMessage.properties }, '*');
+			if (process.env['VSCODE_DEV'] && this.gitpodPreview?.log?.analytics) {
```

Comment:
[nit] Should we always use `this.devMode` or `process.env['VSCODE_DEV']` and only keep one of them in the same file?

Reply:
Is different for the server and the browser, server has access to process.env, browser doesn't